### ISSUE TITLE
Find a way to reduce size of Linux binaries

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#436]
Final Linux builds of Auburn Sounds plug-ins are 4x larger on Linux than they are on Mac / Windows.

**Comment by 0xEAB** [comment]
Could this be related to the *everything is exported by default* on Linux behavior?

**Comment by p0nce** [comment]
Well we are using `-fvisibility=hidden` but perhaps it's not sufficient (on macOS, it isn't)

**Comment by p0nce** [comment]
- [ ] Try `-fvisibility=hidden` AND `-dead_strip`

**Comment by p0nce** [comment]
- [ ] Try `--gc-sections`

**Comment by abaga129** [comment]
I decided to look at this a bit.  I've found some linker flags that can be added to `dub.json` that make some improvement.

```json
"lflags-linux": [
	"-fuse-ld=bfd",
	"-s"
],
```
With convergence + LDC it cuts the binary down from 10.6mb to 8.0mb.  The `-s` flag is the same as calling `strip -s ./binary` on the resulting binary and gives a gain of about 2mb.  `-fuse-ld=bfd` tells it to use bfd rather than gold for the linker which gives a gain of 0.1 to 0.2mb
